src/vivarium/framework/lifecycle.py

```diff
@@ -345,8 +352,8 @@ def check_valid_state(self, method: Callable, permitted_states: List[str]) -> No
             )
 
     def constrain_normal_method(
-        self, method: Callable, permitted_states: List[str]
-    ) -> Callable:
+        self, method: Callable[[Any], Any], permitted_states: list[str]
```

Comment:
All the Callable[[Any], Any] types in this file have extremely broad possible signatures. Some examples I saw: get_randomness_stream, build_lookup_table, ArtifactManager's load, and register_simulant_initializer